mesonbuild/scripts/gettext.py

```diff
@@ -43,19 +45,39 @@ def read_linguas(src_sub):
         print('Could not find file LINGUAS in {}'.format(src_sub))
         return []
 
-def run_potgen(src_sub, pkgname, datadirs, args):
+def run_potgen(src_sub, pkgname, datafilename, datadirs, args):
+    child_env = os.environ.copy()
+    if datadirs:
+        child_env['GETTEXTDATADIRS'] = datadirs
+    ofile = os.path.join(src_sub, pkgname + '.pot')
+
+    if datafilename:
+        with open(datafilename, 'rb') as ifile:
+            files = pickle.load(ifile)
+        tmp_pot = os.path.join(os.path.dirname(datafilename), 'i18n-' + pkgname + '.pot')
+        if os.path.exists(tmp_pot):
+            os.remove(tmp_pot)
+        join_arg = []
+        for f in files:
+            language = ['--language=' + f['language']] if 'language' in f else []
+            keywords = ['--keyword=' + k for k in f['keywords']]
+            flags = ['--flag=' + f for f in f['flags']]
+            subprocess.check_call(['xgettext', '--package-name=' + pkgname, '-p', src_sub,
+                                   '-D', os.environ['MESON_SOURCE_ROOT'], '-o', tmp_pot] + join_arg +
+                                   language + keywords + flags + f['extra_args'] + f['files'],
```

Comment:
__[flake8]__
* [E127] continuation line over-indented for visual indent

<!-- Comment by [SideCI](https://sideci.com) -->